ee/vellum_ee/workflows/display/workflows/base_workflow_display.py

```diff
@@ -408,22 +409,74 @@ def serialize(self) -> JsonObject:
-                "display_data": self.display_context.workflow_display.display_data.dict(),
-                "definition": {
-                    "name": self._workflow.__name__,
-                    "module": cast(JsonArray, self._workflow.__module__.split(".")),
-                },
-                "output_values": output_values,
+        # Serialize workflow-level trigger if present
+        trigger_data: Optional[JsonObject] = self._serialize_workflow_trigger()
+
+        workflow_raw_data: JsonObject = {
+            "nodes": cast(JsonArray, nodes_dict_list),
+            "edges": edges,
+            "display_data": self.display_context.workflow_display.display_data.dict(),
+            "definition": {
+                "name": self._workflow.__name__,
+                "module": cast(JsonArray, self._workflow.__module__.split(".")),
             },
+            "output_values": output_values,
+        }
+
+        if trigger_data is not None:
+            workflow_raw_data["trigger"] = trigger_data
+
+        return {
+            "workflow_raw_data": workflow_raw_data,
             "input_variables": input_variables,
             "state_variables": state_variables,
             "output_variables": output_variables,
         }
 
+    def _serialize_workflow_trigger(self) -> Optional[JsonObject]:
+        """
+        Serialize workflow-level trigger information.
+
+        Returns:
+            JsonObject with trigger data if a trigger is present, None otherwise
+        """
+        # Get all trigger edges from the workflow's subgraphs
+        trigger_edges = []
+        for subgraph in self._workflow.get_subgraphs():
+            trigger_edges.extend(list(subgraph.trigger_edges))
+
+        if not trigger_edges:
+            # No workflow-level trigger defined
+            return None
+
+        # Get the trigger class from the first edge
+        trigger_class = trigger_edges[0].trigger_class
+
+        # Validate that all trigger edges use the same trigger type
+        trigger_type_mapping = get_trigger_type_mapping()
+        for edge in trigger_edges:
+            if edge.trigger_class != trigger_class:
+                raise ValueError(
+                    f"Mixed trigger types not supported. Found {trigger_class.__name__} and "
+                    f"{edge.trigger_class.__name__} in the same workflow."
+                )
+
+        # Get the trigger type from the mapping
+        trigger_type = trigger_type_mapping.get(trigger_class)
+        if trigger_type is None:
+            raise ValueError(
+                f"Unknown trigger type: {trigger_class.__name__}. "
+                f"Please add it to the trigger type mapping in get_trigger_type_mapping()."
+            )
+
+        return {
+            "type": trigger_type.value,
+            "definition": {
+                "name": trigger_class.__name__,
+                "module": cast(JsonArray, trigger_class.__module__.split(".")),
+            },
```

Comment:
[blocking] doesn't appear that we need this `definition` field yet - schema we have so far: https://github.com/vellum-ai/vellum/blob/4d0f6201dde3b345590fe8573e2b44eaf72c7577/django/app/domain_models/types/workflows.py#L2114-L2133

Looks like we also need:
- `id`
- `attributes`, even if we just set to empty array for now